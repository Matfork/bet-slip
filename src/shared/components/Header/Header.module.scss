.Header-component {
  .Header-component__title {
    flex: 1;
  }
}
