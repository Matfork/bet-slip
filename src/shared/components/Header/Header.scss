.Header-component__container {
  .Header-component__title {
    flex: 1;
  }
}
