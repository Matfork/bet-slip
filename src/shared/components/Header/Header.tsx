import { AppBar, IconButton, Toolbar, Typography } from '@material-ui/core';
import React, { useState } from 'react';
import MenuIcon from '@material-ui/icons/Menu';
import DrawerComponent from '../Drawer/Drawer';
import './Header.scss';

const HeaderComponent: React.FC = () => {
  const [showDrawer, setShowDrawer] = useState(false);

  const handleToggleDrawer = (value: boolean): void => {
    setShowDrawer(value);
  };

  return (
    <>
      <AppBar position="static" className="Header-component__container">
        <Toolbar>
          <Typography variant="h6" className="Header-component__title">
            Bet Split
          </Typography>

          <IconButton
            onClick={() => handleToggleDrawer(true)}
            edge="end"
            className="Header-component__hamburger"
            color="inherit"
            aria-label="menu"
          >
            <MenuIcon />
          </IconButton>
        </Toolbar>
      </AppBar>

      <DrawerComponent showDrawer={showDrawer} onToggleDrawer={() => handleToggleDrawer(false)} />
    </>
  );
};

export default HeaderComponent;
