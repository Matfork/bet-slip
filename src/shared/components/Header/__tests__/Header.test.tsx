import { fireEvent, screen } from '@testing-library/react';
import { customRender } from 'test.utilities';
import Header from '../Header';

describe('Header Component', () => {
  it('should test snapshot', () => {
    const component = customRender(<Header />);
    expect(component.container).toMatchSnapshot();
  });

  it('should display title', () => {
    customRender(<Header />);
    const titleEl = screen.queryByText(/Bet Slip/);
    expect(titleEl).toBeInTheDocument();
  });

  describe('toggle drawer', () => {
    const showDrawer = () => {
      const component = customRender(<Header />);

      let titleEl = screen.queryByText(/have any bet selected, try selecting one option/);
      expect(titleEl).toBeNull();

      const button = screen.queryByLabelText(/menu/)!;
      fireEvent.click(button);

      titleEl = screen.queryByText(/have any bet selected, try selecting one option/);
      expect(titleEl).toBeInTheDocument();

      return component;
    };

    it('should display drawer on hamburger icon clicked', () => {
      showDrawer();
    });

    it('should hide drawer on backdrop clicked', () => {
      jest.useFakeTimers();
      expect.assertions(3);

      const component = showDrawer();
      const divBackdrop = component.container.closest('body')!.querySelector('.MuiBackdrop-root')!;

      fireEvent.click(divBackdrop);

      jest.runAllTimers();

      const titleEl = screen.queryByText(/have any bet selected, try selecting one option/);
      expect(titleEl).toBeNull();
    });
  });
});
