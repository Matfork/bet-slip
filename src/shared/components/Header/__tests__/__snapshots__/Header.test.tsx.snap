// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should test snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary Header-component__container MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        class="MuiTypography-root Header-component__title MuiTypography-h6"
      >
        Bet Split
      </h6>
      <button
        aria-label="menu"
        class="MuiButtonBase-root MuiIconButton-root Header-component__hamburger MuiIconButton-colorInherit MuiIconButton-edgeEnd"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</div>
`;
